src/front/wgsl/resolve/mod.rs

```diff
@@ -0,0 +1,1852 @@
+use aho_corasick::AhoCorasick;
+use rustc_hash::{FxHashMap, FxHashSet};
+use std::ops::{Deref, DerefMut};
+
+use crate::front::wgsl::parse::ast;
+use crate::front::wgsl::parse::ast::UnaryOp;
+use crate::front::wgsl::parse::ast::{ExprKind, GlobalDeclKind, Ident, StmtKind, VarDecl};
+use crate::front::wgsl::resolve::dependency::DependencyContext;
+use crate::front::wgsl::resolve::inbuilt::Attribute;
+use crate::front::wgsl::WgslError;
+use crate::{
+    front::wgsl::resolve::{
+        features::{EnabledFeatures, Feature},
+        inbuilt::{
+            reserved_matcher, AccessMode, AddressSpace, AttributeType, Builtin, ConservativeDepth,
+            DepthTextureType, InterpolationSample, InterpolationType, MatType, Matcher,
+            SampledTextureType, SamplerType, ScalarType, StorageTextureType, TexelFormat,
+            ToStaticString, VecType,
+        },
+        inbuilt_functions::InbuiltFunction,
+        index::Index,
+        ir::{CallTarget, DeclDependency, DeclId, InbuiltType, LocalId},
+    },
+    front::wgsl::text::{Interner, Text},
+    BinaryOperator, ImageClass, ScalarKind, Span, StorageAccess, UnaryOperator,
+};
+
+mod dependency;
+pub mod features;
+pub mod inbuilt;
+pub mod inbuilt_functions;
+mod index;
+pub mod ir;
+
+pub struct ResolveContext {
+    dependency_context: DependencyContext,
+    index: Index,
+    reserved_matcher: AhoCorasick,
```

Comment:
It seems like an `FxHashSet<Text>` would be better here than an `AhoCorasick` matcher. You've already hashed the identifier once; it doesn't make sense to make another pass over it.